src/os/README.md

```diff
@@ -104,3 +106,86 @@ inspiration for several decisions in this library.
 - that said, if an operation is extremely efficient and almost always wanted,
   consider doing it natively, automatically. For example, ints in HSL
   `sockaddr` are always in host byte order, not network byte order.
+
+## Hack representation of C data types 
+### Cheat sheet
+
+| C type | Hack type |
+|---|---|
+| Pointers to referential transparent C `struct`s | `vec` |
+| Other C handles or pointers | `HH\Lib\OS\FileDescriptor` or other wrapper classes |
+
+### Default encoding for handles or pointers
+
+The underlying C API commonly exposes some data type as handlers or pointers,
+e.g. `int filedes`, along with functions to create, manipulate, and destroy
+them. By default, handlers and pointers should be wrapped in HSL as Hack
+classes.
+
+### Encoding referential transparent C pointers
+
+However, as an optimization, pointers to referential transparent C `struct`s can
+be encoded as a garbage-collectable mirror in Hack, which creates and destroys
+the underlying C data type on demand. 
+
+Technically, the terminology "referential transparent" should be interpreted as
+"referential transparent except for memory allocation" in C, while its
+corresponding Hack mirror is strictly referential transparent, which means there
+is no behavior change when the user replaces a variable of the referential
+transparent mirror with the expression creating the variable, and vice-versa.
+
+#### The definition of referential transparent C pointer
+
+- A referential transparent C pointer type should be associated with a set of
+utility functions, including a constructor, a destructor and several setters.
+These utility functions must not perform any global side effects other than
+allocating memory that will be freed in the destructor.
```

Comment:
File descriptors are used as an example, but this is not the case for file descriptors.

Reply:
I don't understand this comment, because I think file descriptors are not referential transparent C pointers, and I did not use file descriptors as an example for the definition of referential transparent C pointers.